### PR TITLE
Create weather-app using API calls and javascript async-await and other ES6 features 

### PR DESCRIPTION
**Pull request by adejam**
# Project: WEATHER-APP

The project follows the guideline of [the odin project](https://theodinproject.com/courses/javascript/lessons/weather-app) to build a weather app using javascript for DOM manipulation and interaction. It uses javascript functions.It uses Async-await and API calls, It uses webpack. It also uses Bootstrap for user interface design.

- It demonstrates the use of API calls and javascript Async-Await and other javascript features as follows:
  - DOM manipulation APIs (i.e: `querySelector()`, `event Listeners` and more);
  - Arrow function.
  - Spread Operator
  - Object literal
  - Ternary Operators and more ES6 based syntax.

- Uses Semantic html
- Uses Webpack
- Uses Bootstrap for user interface
- Uses stylint and prettier for formatting.
- Uses linter to check for errors in javascript
- Uses Jest for testing

### CODE REVIEW

File: src/js/converter.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const toFarenheit = celsius => Math.floor(celsius * (9 / 5) + 32);
+
+const toMph = speed => Math.floor(speed / 1.609);
+
+const converter = value => ({
+  farenheit: toFarenheit(value),
+  mph: toMph(value),
+});
```
Comment (bolah2009) [inline src/js/converter.js:8]: Can you refactor this to use factory function (see [Odin's Factory function lesson](https://www.theodinproject.com/courses/javascript/lessons/factory-functions-and-the-module-pattern)?
This way the functions will not always execute for the passed argument even when it's not needed.

File: src/js/datebuilder.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    'December',
+  ];
+
+  const days = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
+  const currentDay = days[date.getDay()];
+  const currentDate = date.getDay();
+  const month = months[date.getMonth()];
+  const year = date.getFullYear();
+  return `${currentDay} ${currentDate} ${month} ${year}`;
+};
```
Comment (bolah2009) [inline src/js/datebuilder.js:23]: Can you add test for this too?

Also, not that there's a bug with this function see below screenshot.

image

File: src/js/displayresults.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  const humidity = Math.round(weather.main.humidity);
+  const notification = document.querySelector('#notification');
+
+  notification.classList.add('d_none');
+
+  const tempValFar = converter(tempVal).farenheit;
+  const minValFar = converter(minVal).farenheit;
+  const maxValFar = converter(maxVal).farenheit;
+  const speedMph = converter(speed).mph;
+  const main = document.querySelector('#main');
```
Comment (bolah2009) [inline src/js/displayresults.js:23]: Can we refactor this logic into a function and move it out of here? 

Let's create a function that will clean the data, process it to what you want to show to your user and return a structured data for it to be populated and displayed to the DOM.

Let's add test for the refactored function.

File: src/js/setposition.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import api from './api';
+import getWeather from './getweather';
+
+const setPosition = position => {
+  const { latitude } = position.coords;
+  const { longitude } = position.coords;
```
Comment (bolah2009) [inline src/js/setposition.js:6]: Can we destructure both variables at once?

File: src/js/unitchanger.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+/* eslint no-unused-expressions: ["error", { "allowTernary": true }] */
+import changeUnits from './changeunits';
+
+const unitChanger = ({ target: { id, classList } }) => {
+  if (classList.contains('changer')) {
+    const toImperial = ['.celcius', '.farenheit', '.toMetric', '.toImperial'];
+    const toMetric = ['.farenheit', '.celcius', '.toImperial', '.toMetric'];
+    id === 'toImperial' ? changeUnits(...toImperial) : changeUnits(...toMetric);
+  }
```
Comment (bolah2009) [inline src/js/unitchanger.js:9]: I don't think we need to use ternary operator here. Let's use nested if instead

**Review by bolah2009 — APPROVED** [review]
It's a 👍🏾 from me. 

Please make the suggested changes before merging.

**Comment by adejam** [comment]
Thanks for the review.

File: src/js/converter.js (hunk 1 above)
Comment (adejam) [reply src/js/converter.js:8]: I'll check the lessonand make necessary adjustments

File: src/js/datebuilder.js (hunk 2 above)
Comment (adejam) [reply src/js/datebuilder.js:23]: I'll ensure to debug this

File: src/js/displayresults.js (hunk 3 above)
Comment (adejam) [reply src/js/displayresults.js:23]: I'll improve on this

**Pull request merged**